### ISSUE TITLE
Suggestion and Question

### ISSUE THREAD

**Issue by xorander00** [Thefrank/jellyfin-server-freebsd#11]
Thanks for doing this, looks good! :)  
  
**Suggestion**  
Instead of symlinking the sqlite lib, it would probably be better to use libmap.  Even though it should be used sparingly, it's an ideal use-case for a situation like this.  It's as follows:  
  
```
# /usr/local/etc/libmap.d/jellyfin.conf
libe_sqlite3.so libsqlite3.so
```
  
...and that's it, viola! Runtime linker now automatically maps any requests for libe_sqlite3.so to libsqlite3.so :)  
  
**Question** 
Are there specific build from source instructions available? It would be nice to have a reference doc for building from source. Once I have a bit more time, I'd be willing to create a port for this and submit it to the tree. Instead of having to figure it out from scratch, having an initial starting point would save a bunch of time and would be more likely to get done sooner.  Also, I'm on FreeBSD 13.0 (1300522 more specifically) and even though this seems to launch fine on it, I know it's built for 12.x (and thus pkg won't want to install it, unless forced).
  
Thanks again! :)

**Comment by Thefrank** [comment]
I will look into using libmap when I get a bit more time. When I first looked to making a package for this I noticed libmap had a number of warnings and caveats with it so I opted to just symlink instead. It does however look like an ideal solution...when I get more time to test it :)

As for building, a source build will not work in ports tree* as the "build" segment uses `dotnet restore` which...uses the internet to restore packages. With a working dotnet 5 or 6 SDK for FreeBSD, neither `jellyfin-web` or `jellyfin` require code changes.

- `jellyfin-web` can be built from the `./build.sh`
- `jellyfin` needs to use `dotnet publish` to avoid making code changes (e.g., `./build.sh` checks for `dpkg` which FreeBSD does not have)
- packaging is done with `tar`
- pkg is made with ports system as `NO_BUILD` (This is also your best bet for getting it into the ports tree see: `radarr` as dotnet5 https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=259194  or `prowlarr` already in ports https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=259196)

Finally, I build for whatever TrueNAS is currently using as the bulk of the users seem to come from there. If there is increased demand from base FreeBSD users I can setup something for different versions of FreeBSD :)

*I did try and look into a way of getting the build system to download and unpack the NuGet files that dotNET uses during the `fetch` and `extract` phases. The download is fine but I have no idea how `dotnet` unpacks NuGet files. There is a large difference between how `dotnet` unpacks a NuGet and how `tar` unpacks it.

**Comment by xorander00** [comment]
Yeah, libmap is pretty nice. I've had to use it in the past for a few other things too (e.g. phantomjs when an app I was using still required it long after it was deprecated) and it tends to save a decent amount of hassell.  For what it's worth, on my 13.x box it seems to be working for Jellyfin, though I can't speak for 12.x/TrueNAS/etc.  
  
> As for building, a source build will not work in ports tree* as the "build" segment uses dotnet restore which...uses the internet to restore packages.  
  
Ports can have distribution files specified (or upstream URLs from which to fetch additional files).  It was something I figured I'd have to do for this anyway.  
  
> jellyfin needs to use dotnet publish to avoid making code changes (e.g., ./build.sh checks for dpkg which FreeBSD does not have)  
  
So what would happen in this case is a patch would be generated and bundled with the port, which is what I do usually for my own.  Avoiding code changes would be ideal, but if necessary, I wouldn't mind making some basic changes & then generating the patch and specifying the apply in the port manifest.  
  
> The download is fine but I have no idea how dotnet unpacks NuGet files. There is a large difference between how dotnet unpacks a NuGet and how tar unpacks it.  
  
I can take care of this part, though it has been a while since I've touched dotnet and I'll have to do a quick refresher. They're essentially zip files with their own dotnet-specific definition files (standard .config files, etc), at least they were last I looked.  
  
> I totally forgot about Skia! This was extremely painful to get built due to the lack of google's tools being unaware of FreeBSD. I have a whole separate jail preserved incase I need to build a newer version of it.  
  
Don't even get me started on Google stuff lol.  A majority of their projects are pretty Linux-specific, even when they don't need to be.  On top of that, even the stuff that's Linux-specific is put together in such a hacky way half the time that I have to roll my eyes. I've had to go through my share of code to get it to build on FreeBSD.  
  
I'll take a look further next week and see what I can dig up in general. Let me know if you have any thoughts.
